config/v2/resolvers.go

```diff
@@ -0,0 +1,188 @@
+package v2
+
+import (
+	"github.com/chanzuckerberg/fogg/config/v1"
+)
+
+// lastNonNil, despite its name can return nil if all results are nil
+func lastNonNil(getter func(Common) *string, commons ...Common) *string {
+	var s *string
+	for _, c := range commons {
+		t := getter(c)
+		if t != nil {
+			s = t
+		}
+	}
+	return s
+}
+
+// lastNonNilInt, despite its name can return nil if all results are nil
+func lastNonNilInt64(getter func(Common) *int64, commons ...Common) *int64 {
+	var s *int64
+	for _, c := range commons {
+		t := getter(c)
+		if t != nil {
+			s = t
+		}
+	}
+	return s
+}
+
+// lastNonNilStringSlice, despite its name can return nil if all results are nil
+func lastNonNilStringSlice(getter func(Common) []string, commons ...Common) []string {
+	var s []string
+	for _, c := range commons {
+		t := getter(c)
+		if t != nil {
+			s = t
+		}
+	}
+	return s
+}
+
+// ResolveRequiredString will resolve the value and panic if it is nil. Only to be used after validations are run.
+func ResolveRequiredString(getter func(Common) *string, commons ...Common) string {
```

Comment:
I feel like for these the `Must...` is a kind of a go convention. Maybe `MustResolveString`

Reply:
Good idea. This diff is already getting big, so I would like to do this as a followup

Reply:
https://github.com/chanzuckerberg/fogg/issues/250